### PR TITLE
chore(ci): fixed ci release and removed Node.js 16

### PR DESCRIPTION
**Pull request by carlocorradini**

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -133,8 +133,6 @@ jobs:
           npm run build:benchmarks
 
       - name: Check
-        # only run on newest node version
```
Comment (MichalLytek) [inline .github/workflows/check.yml:136]: Why removing this? Does it make sense to run this on all node verions?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -17,7 +15,7 @@ jobs:
   release:
     name: Release package on NPM
     runs-on: ubuntu-latest
-    if: github.event.workflow_run.conclusion == 'success' && startsWith(github.event.workflow_run.head_branch, 'refs/tags/v')
+    if: github.event.workflow_run.conclusion == 'success' && github.ref_name == 'master' && startsWith(github.event.workflow_run.head_branch, 'v')
```
Comment (MichalLytek) [inline .github/workflows/release.yml:18]: What does it fix and how?

File: .github/workflows/check.yml (hunk 1 above)
Comment (carlocorradini) [reply .github/workflows/check.yml:136]: No, it's better to run everything on all Node.js versions.
Who knows that something will break :/

File: .github/workflows/release.yml (hunk 2 above)
Comment (carlocorradini) [reply .github/workflows/release.yml:18]: Run this workflow only if the previous workflow (check) is successful
AND
If running on master
AND
If the head_branch starts with v (tag that has already been checked in check)

**Review by MichalLytek — APPROVED** [review]
Let's see

**Pull request merged**